**1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.**

X +- z(alpha/2) * sigma / sqrt(n)

In [1]:
import numpy as np
import scipy.stats as stats
M = 80
n = 256
sigma = 16
Z = stats.norm.ppf(0.975)

print(M - Z * sigma/np.sqrt(n))
print(M + Z * sigma/np.sqrt(n))

78.04003601545995
81.95996398454005


Доверительный 95% интервал для оценки мат. ожидания = [78.04, 81.96]

**2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные:**
**6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1**

**Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.**

In [2]:
import numpy as np
import scipy.stats as stats


def confidence_interval(x, alpha=0.05):
    n = len(x)
    mu = np.mean(x)
    sigma = np.std(x, ddof=1)
    t = stats.t.ppf(1 - (alpha / 2), n - 1)
    return round(mu - t * (sigma / np.sqrt(n)), 3), round(mu + t * (sigma / np.sqrt(n)), 3)


x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

confidence_interval(x)

(6.268, 6.912)

Доверительный интервал 95% для оценки истинного значения = [6.268, 6.912]

**2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные:**
**6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1**

**Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.**

In [3]:
import numpy as np 
import scipy.stats as stats 

def lu(x, y):
    lengths = (len(x), len(y))
    delta = np.mean(x) - np.mean(y)
    dispersion = 0.5 * (np.var(x, ddof=1) + np.var(y, ddof=1))
    s_e = np.sqrt((dispersion / lengths[0]) + (dispersion / lengths[1]))
    t = stats.t.ppf(0.975, 2 * (lengths[0] - 1))
    return delta - t * s_e, delta + t * s_e


x = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
y = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])

lu(x, y)

(-10.068418034506857, 6.268418034506846)

Доверительный интервал 95% для оценки разности среднего роста родителей и детей [-10.07, 6.27]. Так как в интервале имеется 0, значит принимаем нулевую гипотезу и на основании этого делаем вывод, что статистических различий между ростом матерей и дочерей нет.